walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,136 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Iterator;
+
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    private ArrayList<TimeRange> availableTimes = new ArrayList<>();
```

Comment:
This could probably be ok without being a class member, look into if you can create locally in the method instead and pass it around via method params.

Reply:
I thought about that, but was a bit worried about it being copied as its passed between the methods. I realize that since it's working with a fairly small data size that might not be a concern, but would it be good practice to keep it as a class member if  I know it's going to take up a lot of space?

Reply:
> but was a bit worried about it being copied as its passed between the methods

Really good thinking about this.  It definitely would be a problem for some languages (C++ for example).  I'm not 100% positive with Java, but I believe containers are always passed as reference, so there would not be a copy. 

>  but would it be good practice to keep it as a class member if I know it's going to take up a lot of space

Also a really good question, it shows you're thinking about the structure of your class a bit more.  In general think of scope and encapsulation when trying to decide if something should be a member or a local variable.  ie. create as a member if keeping the state/value for as long as the class lives is needed.  For example if you have a Tree class, you'd definitely want your Tree Nodes map/list to be a member, since you'll need your Nodes to live for as long as the class itself lives.

In this specific scenario, since you generate a new availableList for each time you call query(), there's no need to keep hold of the state/value for longer than the method call.  Meaning generating the list locally within the method, or passing a list in as a param to the query method would be sufficient